docs/examples/sentence_classification.md

```diff
@@ -0,0 +1,344 @@
+
+
+```python
+from river import feature_extraction
+from river import naive_bayes
+from river import metrics
+
+def extract_body(x):
+    """Extract the body of the sms."""
+    return x['body']
+
+X_y = datasets.SMSSpam()
+
+model = extract_body | feature_extraction.TFIDF() | naive_bayes.BernoulliNB(alpha = 0)
+
+metric = metrics.ROCAUC()
+
+cm = metrics.ConfusionMatrix()
+
+for x, y in X_y:
+    
+    y_pred = model.predict_one(x)
+    
+    if y_pred is not None:
+        
+        metric.update(y_pred=y_pred, y_true=y)
+        
+        cm.update(y_pred=y_pred, y_true=y)
+    
+    model.learn_one(x, y)
+    
+metric
+```
+
+
+
+
+    ROCAUC: 0.929296
+
+
+
+The confusion matrix:
+
+
+```python
+cm
+```
+
+
+
+
+               False    True
+       False    4809      17
+        True     103     644
+
+
+
+The results are quite good with this first model.
+
+Since we are working with an unbalanced dataset, we can use the `imblearn` module to artificially increase the number of spam in our dataset. For more information about the `imblearn` module, you can find a dedicated tutorial [here](https://riverml.xyz/latest/examples/imbalanced-learning/).
+
+
+```python
+from river import imblearn
+
+X_y = datasets.SMSSpam()
+
+model = extract_body | feature_extraction.TFIDF() | imblearn.RandomUnderSampler(
+    classifier=naive_bayes.BernoulliNB(alpha = 0),
+    desired_dist={0: 0.5, 1: 0.5},
+    seed=42
+)
+
+metric = metrics.ROCAUC()
+
+cm = metrics.ConfusionMatrix()
+
+for x, y in X_y:
+    
+    y_pred = model.predict_one(x)
+    
+    if y_pred is not None:
+        
+        metric.update(y_pred=y_pred, y_true=y)
+        
+        cm.update(y_pred=y_pred, y_true=y)
+    
+    model.learn_one(x, y)
+    
+metric
+```
+
+
+
+
+    ROCAUC: 0.951728
+
+
+
+The `imblearn` module improved our results. Not bad! We can visualize the pipeline to understand how the data is processed.
+
+The confusion matrix:
+
+
+```python
+cm
+```
+
+
+
+
+               False    True
+       False    4624     201
+        True      41     706
+
+
+
+
+```python
+model.draw()
+```
+
+
+
+
+    
+![svg](sentence_classification_files/sentence_classification_14_0.svg)
+    
+
+
+
+Now let's try to use logistic regression to classify messages. I will use different tips to make my model perform better. As in the previous example, I artificially increase the number of spam. The logistics regression will be fed from a tf-idf.
+
+
+```python
+from river import linear_model
+from river import optim
+from river import preprocessing
+
+X_y = datasets.SMSSpam()
+
```

Comment:
I think using a `Normalizer` instead of a `StandardScaler` is more appropriate. At least this is what scikit-learn recommends. Indeed the goal is to normalize the length of documents, and not necessarily the distribution of a feature's occurrences. In other words it should be better to normalize row-wise rather than column-wise.